src/osd/OSDMap.cc

```diff
@@ -4641,6 +4641,7 @@ int OSDMap::calc_pg_upmaps(
       // look for remaps we can un-remap
       vector<pair<pg_t,
         mempool::osdmap::vector<pair<int32_t,int32_t>>>> candidates;
+      candidates.reserve(tmp.pg_upmap_items.size());
```

Comment:
what's the scale of `tmp.pg_upmap_items` under most circumstances? presumably it contains the pgs that are remapped.

Reply:
may vary, from dozens to thousands, depending on the scale of cluster itself and how imbalanced the distribution of PGs are.